[feat] 간병인 간병지원 기능

leejs96 opened the pull request.

leejs96 committed:
[feat] 간병인 간병지원 기능

DojinP commented on src/main/java/com/suitecare/suitecare/api/reservation/controller/ReservationController.java:
검색해보니 rest api 명세에서 url 의 작성은 행위의 주체 -> 행위 -> 대상 순으로 작성되는 거 같아요.
아래와 같은 mate가 예약을 검색하는 api 의 경우,
'/mate/search/reservation' 혹은 '/search/reservation' 이 되면 직관적이고 좋을 것 같습니다.

DojinP replied:
위 내용과 무관하지만, Controller 의 메소드와 Service 메소드명이 동일한 로직임에도 다른데 통일해주시면 좋을 것 같습니다.

DojinP commented on src/main/java/com/suitecare/suitecare/api/reservation/dto/ApplyReservationRequestDTO.java:
현재 MySQL에 모든 테이블 id 는 BIGINT 형으로 정의되어 있습니다.
BIGINT 에 대응하는 정수 자료형은 Long 타입이므로 Integer -> Long 타입으로 변경해주세요.

DojinP commented on src/main/java/com/suitecare/suitecare/config/SecurityConfig.java:
url 수정이 필요해보입니다. Controller 에 남긴 코멘트 참고하셔서 수정해주시고, 난해한 부분이 있다면 말씀해주세요.

DojinP commented on src/main/resources/mapper/ReservationMapper.xml:
이건 테이블 명 수정 요청 드렸었는데, 한 번 확인해주세요!

DojinP reviewed (changes requested):
코멘트 참고하셔서 수정 부탁드려요.

leejs96 committed:
[feat/refactor] 간병 지원시 간병인 이력서 없는 경우, 이미 지원한 내역인 경우 기능 추가 / url변경, 변수명 변경

- 간병인 이력서가 없는 경우 return 0
- 이미 지원한 간병인 경우 return 2
- 테이블명 변경(matesofreservation -> applicantofreservation)
- url변경 mate/search -> search/reservation

leejs96 committed:
[refactor] 주석처리취소, 변수 타입 변경

leejs96 committed:
[refactor] 간병인 이력서 여부, 지원여부 메서드 따로 뺌

leejs96 committed:
[refactor] 변수 타입 실수 변경 / DB 테이블명 변경에 따른 sql문 수정

leejs96 committed:
[fix] 이미 지원한 경우로 나오는 부분 수정

- 이미 지원한 경우 null로 판단하는 것이 아니라 결과값이 0인지 아닌지 확인했어야함

DojinP commented:
확인했습니다
